csrgen.py

```diff
@@ -0,0 +1,190 @@
+        return False
+    else:
+        return file
+
+
+def check_domain(input):
+    if check_file(input):
+        return input, True
+    else:
+        domain_template = re.compile(
+            "^(?=.{1,255}$)(?!-)[A-Za-z0-9\-]{1,63}(\.[A-Za-z0-9\-]{1,63})*\.?(?<!-)$")
+        if domain_template.match(input):
+            return input.lower()
+        else:
+            raise argparse.ArgumentError(
+                "Unable to determine if this is a domain or a file. Please check your input.")
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description=description,
+                                     formatter_class=argparse.RawTextHelpFormatter)
+    subparser = parser.add_subparsers(dest="command")
+
+    csr = subparser.add_parser("csr", help="Create CSR",
+                               formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    csr.add_argument('-k', "--key", type=check_file, help='Specify a key')
+    csr.add_argument('-d', "--domain", type=check_domain, required=True,
+                     help="Specify a domain or a file with list of domains")
+    csr.add_argument("-o", "--org", help="Organization Name")
+
+    config = subparser.add_parser("config", help="Create config file")
+    config.add_argument("-c", "--country", help="Country short code")
+    config.add_argument("-l", "--locality", help="Locality/City")
+    config.add_argument("-s", "--state", help="State Name")
+    config.add_argument("-o", "--org", help="Organization Name", required=True)
+
+    if parser.parse_args().command:
+        return parser.parse_args()
+    else:
+        parser.print_help()
+        exit()
+
+
+def gen_csr_config(args, config_file):
+    tm = env.get_template(config_file)
+    print(config_file)
+    SANS = list()
+    print("Generating CSR configuration..")
+    if type(args.domain) is tuple:
+        domain_file, is_file = args.domain
+        with open(domain_file) as fp:
+            domain = fp.readline()
+            count = 0
+            while domain:
+                print(domain)
+                if count == 0:
+                    CN = domain.strip().lower()
+                    domain = fp.readline()
+                    count += 1
+                elif 1 <= count <= 100:
+                    SANS.append(domain.strip().lower())
+                    domain = fp.readline()
+                    count += 1
+                else:
+                    print("Too many domains! Please limit to 100")
+                    exit(1)
+            print(CN)
+    else:
+        domain = args.domain
+        CN = domain.strip().lower()
+
+    generated_csr_config = open(args.domain + '_csr_config', 'w')
+    generated_csr_config.write(tm.render(cn=CN, sans=SANS))
+    generated_csr_config.close()
+
+    return str(args.domain + '_csr_config')
+
+
+def main(args):
+    if args.command == 'csr':
+        # Look for csrgen config files
+        config_file_search = fnmatch.filter(os.listdir(), "*_gen_config")
```

Comment:
I see `_gen_config` in quite a few places. Consider replacing it with a named constant for maintainability.